### ISSUE TITLE
Blocks checkout removes unavailable coupon code without informing the shopper and processes payment

### ISSUE THREAD

**Issue by pierorocca** [woothemes/woocommerce#44906]
> [!NOTE]  
> this might be unblocked by the work we'll be doing on https://github.com/woocommerce/woocommerce/issues/44231

### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

If two shoppers have both applied a limited use coupon code and try to check out:

-  the first shopper will get the discount and will successfully place the order
- the second shopper will have the discount removed silently and will successfully place the order at a non-discounted amount

https://github.com/woocommerce/woocommerce/assets/109928953/bfc253dd-0eb7-4cfd-b7c1-df2773405182

### Expected behavior

Shortcode like behavior which alerts the shopper with an error, removes the coupon code, updates the total and allows the shopper to recover.

![image](https://github.com/woocommerce/woocommerce/assets/109928953/9d4bfb47-b478-4a6f-a1c3-f82698951e1e)

### Actual behavior

The order is placed at the non-discounted rate and the shopper is unaware that the coupon code was removed.

### Steps to reproduce

See video recording. Tsubaki theme in use. Chrome on Mac OS.

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://proccaatomic.wpcomstaging.com
Site address (URL): https://proccaatomic.wpcomstaging.com
WC Version: 8.6.1
REST API Version: ✔ 8.6.1
WC Blocks Version: ✔ 11.8.0-dev
Action Scheduler Version: ✔ 3.7.1
Log Directory Writable: ✔
WP Version: 6.4.3
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.1.27
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 8.4.0
OpenSSL/1.1.1n

SUHOSIN Installed: –
MySQL Version: 10.4.25-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.6.1
WC Database Prefix: wp_
Total Database Size: 30.63MB
Database Data Size: 24.07MB
Database Index Size: 6.56MB
wp_woocommerce_sessions: Data: 0.17MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.06MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.33MB + Index: 0.28MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.14MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.27MB + Index: 0.19MB + Engine InnoDB
wp_afwc_campaigns: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_afwc_commission_plans: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_afwc_hits: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_afwc_payouts: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_afwc_payout_orders: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_afwc_referrals: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.20MB + Index: 0.09MB + Engine InnoDB
wp_gla_attribute_mapping_rules: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gla_budget_recommendations: Data: 0.22MB + Index: 0.14MB + Engine InnoDB
wp_gla_merchant_issues: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_gla_shipping_rates: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_gla_shipping_times: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_jetpack_sync_queue: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_custom_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_dynamic_segment_filters: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_feature_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_mapping_to_external_entities: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_newsletter_links: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_newsletter_option: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_option_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_posts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_segment: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_templates: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_scheduled_tasks: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_scheduled_task_subscribers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_segments: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_sending_queues: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_settings: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_statistics_bounces: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_statistics_clicks: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_statistics_forms: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_statistics_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_statistics_opens: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_mailpoet_statistics_unsubscribes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_statistics_woocommerce_purchases: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_mailpoet_stats_notifications: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_subscribers: Data: 0.02MB + Index: 0.13MB + Engine InnoDB
wp_mailpoet_subscriber_custom_field: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_subscriber_ips: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_subscriber_segment: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_subscriber_tag: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_tags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_user_agents: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_user_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 2.11MB + Index: 0.09MB + Engine InnoDB
wp_postmeta: Data: 7.52MB + Index: 1.83MB + Engine InnoDB
wp_posts: Data: 3.08MB + Index: 0.09MB + Engine InnoDB
wp_snippets: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.08MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.16MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.13MB + Index: 0.09MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.08MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.08MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availability: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availabilitymeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_booking_relationships: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.09MB + Index: 0.13MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.28MB + Index: 0.67MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.08MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_bundle_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.05MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.05MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_points_rewards_user_points_log: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_bundled_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_bundled_items: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_gc_activity: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_gc_cards: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_gc_cardsmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wpml_mails: Data: 4.52MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

amp_validated_url: 8
attachment: 42
customize_changeset: 12
jetpack_migration: 2
jp_img_sitemap: 1
jp_pay_product: 1
jp_sitemap: 1
jp_sitemap_master: 1
mailpoet_page: 1
nav_menu_item: 5
page: 14
post: 2
product: 32
product_variation: 12
revision: 30
shop_coupon: 4
shop_order: 234
shop_subscription: 7
wcpf_item: 5
wcpf_project: 1
wp_global_styles: 6
wp_navigation: 1
wp_template: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (17) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.1
Code Snippets: by Code Snippets Pro – 3.6.2
Crowdsignal Forms: by Automattic – 1.7.0
WordPress.com Editing Toolkit: by Automattic – 4.13440
Gutenberg: by Gutenberg Team – 17.7.0
Jetpack: by Automattic – 13.2-a.7
Layout Grid: by Automattic – 1.8.4
Cron Jobs: by Ariel – 1.2.9
Mailchimp for WooCommerce: by Mailchimp – 3.6
Page Optimize: by Automattic – 0.5.5
Crowdsignal Polls & Ratings: by Automattic
Inc. – 3.1.1

WooCommerce Payments Dev Tools: by Automattic –
WooPayments: by Automattic – 7.2.0
WooCommerce PayPal Payments: by WooCommerce – 2.5.3
WooCommerce Shipping & Tax: by WooCommerce – 2.5.1
WooCommerce: by Automattic – 8.6.1
WP Mail Logging: by WP Mail Logging Team – 1.12.0

### Inactive Plugins (13) ###

Affiliate For WooCommerce: by StoreApps – 7.3.0 (update to version 7.5.0 is available)
AMP: by AMP Project Contributors – 2.5.2
Classic Editor: by WordPress Contributors – 1.6.3
Google Listings and Ads: by WooCommerce – 2.5.18
Product Filters for WooCommerce: by WooCommerce – 1.4.20 (update to version 1.4.21 is available)
WooCommerce Blocks: by Automattic – 11.7.0
WooCommerce Bookings: by WooCommerce – 2.0.9
WooCommerce EU VAT Number: by WooCommerce – 2.9.1
WooCommerce Points and Rewards: by WooCommerce – 1.7.48 (update to version 1.7.49 is available)
WooCommerce Stripe Gateway: by WooCommerce – 7.9.3
Woo Gift Cards: by Woo – 1.16.10
Woo Product Bundles: by Woo – 6.22.6 (update to version 6.22.7 is available)
Woo Subscriptions: by WooCommerce – 6.0.0

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

WP.com Site Helper: by  –

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: booking (booking)
bundle (bundle)
external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: ✔
Enforce Approved Product Download Directories: ✔
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: ✔

### WC Pages ###

Shop base: #378 - /shop/
Cart: #284 - /cart-blocks/
Checkout: #13 - /checkout/
My account: #517 - /my-account-2/
Terms and conditions: #782 - /terms-page/

### Theme ###

Name: Tsubaki
Version: 1.3.0
Author URL: https://automattic.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: /wordpress/plugins/woocommerce/8.6.1/templates/block-notices/error.php
/wordpress/plugins/woocommerce/8.6.1/templates/block-notices/notice.php
/wordpress/plugins/woocommerce/8.6.1/templates/block-notices/success.php
/wordpress/themes/premium/tsubaki/parts/mini-cart.html
/wordpress/themes/premium/tsubaki/templates/archive-product.html
/wordpress/themes/premium/tsubaki/templates/product-search-results.html
/wordpress/themes/premium/tsubaki/templates/single-product.html
/wordpress/themes/premium/tsubaki/templates/taxonomy-product_cat.html
/wordpress/themes/premium/tsubaki/templates/taxonomy-product_tag.html

### WooPayments ###

Version: 7.2.0
Connected to WPCOM: Yes
WPCOM Blog ID: 209209871
Account ID: acct_1LT92K2EF8r2i5A2
Payment Gateway: Enabled
Test Mode: Enabled
Enabled APMs: card,affirm,afterpay_clearpay
WooPay: Enabled (product,checkout,cart)
WooPay Incompatible Extensions: No
Apple Pay / Google Pay: Enabled (product,cart,checkout)
Fraud Protection Level: basic
Multi-currency: Enabled
Public Key Encryption: Disabled
Auth and Capture: Enabled
Documents: Disabled
Logging: Enabled

### Subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
Subscriptions Live URL: https://proccaatomic.wpcomstaging.com
Subscriptions-core Library Version: 6.7.1
Subscription Statuses: wc-active: 3
wc-on-hold: 3
wc-pending: 1

WooCommerce Account Connected: ✔ Yes
Active Product Key: ❌ No

### Store Setup ###

Country / State: United States (US) — California

### Subscriptions by Payment Gateway ###

WooPayments: wc-active: 3
wc-on-hold: 2

### Payment Gateway Support ###

WooPayments: products
refunds
multiple_subscriptions
subscription_cancellation
subscription_payment_method_change_admin
subscription_payment_method_change_customer
subscription_payment_method_change
subscription_reactivation
subscription_suspension
subscriptions
gateway_scheduled_payments
tokenization
add_payment_method

Check payments: products
Cash on delivery: products
Direct bank transfer: products
WooPayments (Affirm): products
refunds
multiple_subscriptions
subscription_cancellation
subscription_payment_method_change_admin
subscription_payment_method_change_customer
subscription_payment_method_change
subscription_reactivation
subscription_suspension
subscriptions
gateway_scheduled_payments
tokenization
add_payment_method

WooPayments (Afterpay): products
refunds
multiple_subscriptions
subscription_cancellation
subscription_payment_method_change_admin
subscription_payment_method_change_customer
subscription_payment_method_change
subscription_reactivation
subscription_suspension
subscriptions
gateway_scheduled_payments
tokenization
add_payment_method

### WooCommerce PayPal Payments ###

Onboarded: ✔
Shop country code: US
WooCommerce currency supported: ✔
Advanced Card Processing available in country: ✔
Pay Later messaging available in country: ✔
Webhook status: –
PayPal Vault enabled: –
ACDC Vault enabled: –
Logging enabled: ✔
Reference Transactions: ✔
Used PayPal Checkout plugin: –
Subscriptions Mode: Disabled

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
product-linked
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-pre-publish-modal
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2024-02-23 20:01:32 -08:00
Options: ✔
Notes: 132
Onboarding: completed

### Action Scheduler ###

Canceled: 4
Oldest: 2024-02-14 11:11:11 -0800
Newest: 2024-02-22 09:29:12 -0800

Complete: 297
Oldest: 2024-01-23 01:00:27 -0800
Newest: 2024-02-22 09:32:10 -0800

Failed: 29
Oldest: 2022-08-04 14:05:41 -0700
Newest: 2024-02-16 12:55:35 -0800

Pending: 6
Oldest: 2024-02-23 02:10:33 -0800
Newest: 2024-11-06 19:22:01 -0800

### Status report information ###

Generated at: 2024-02-22 12:56:40 -08:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by opr** [comment]
Hi @pierorocca I was able to reproduce this.

The cart update route silently removes coupons when the address is changed while an invalid coupon is present. I think we'll need to add some errors on the response if this happens. Not sure right now if there's a mechanism in place to do this but we can look into it as part of fixing the issue. Thanks Piero!

I am applying the high priority label here as it is likely to affect high volume stores that issue limited use coupons.

**Comment by pierorocca** [comment]
Thanks @opr. Do you expect this one to be picked up this sprint given the potential high impact and since payment would be made at the higher price?

**Comment by opr** [comment]
@pierorocca I don't know if it's likely unfortunately, @ralucaStan and @pmcpinto would be better to answer the question of priority.

**Comment by pmcpinto** [comment]
Just to clarify: is this only happening when 2 or more shoppers try to apply the same coupon simultaneously?

**Comment by pierorocca** [comment]
> Just to clarify: is this only happening when 2 or more shoppers try to apply the same coupon simultaneously?

That's correct @pmcpinto. Picture a scenario where a merchant is running a promo and first 50 get the discount, the promo drives heavy traffic, one person successfully gets the discount and depending on the timing/session duration many others think they're getting the discount but instead get a successful order unknowingly placed at the higher price.

Shortcode is working as expected.

**Comment by ralucaStan** [comment]
I was able to replicate this. @pmcpinto IMO this should go to the out-of-the-box epic. The question is when should we work on it?

**Comment by pmcpinto** [comment]
Most stores using the block have a low number of daily checkout sessions. Therefore, this seems to be an edge case. But it would be good to try to include a fix in 8.9 (code freeze: April 24th).

**Comment by pmcpinto** [comment]
I'll remove the high prio label, as we'll only work on this in the next cycle.

**Comment by ralucaStan** [comment]
There is a possibility this might be unblocked by the work we'll be doing on https://github.com/woocommerce/woocommerce/issues/44231. I'll mark this as blocked until we have a direction on stock handling.

**Comment by senadir** [comment]
Code that checks if there are any errors is here
https://github.com/woocommerce/woocommerce//blob/d18a7225018d90b5f1ab06ff0a7bca6d0b969f0b/plugins/woocommerce/src/StoreApi/Utilities/OrderController.php#L224-L252

It doesn't seem to run correctly, possibly due to this

https://github.com/woocommerce/woocommerce-blocks/pull/11860/

**Comment by mikejolley** [comment]
So far I've found if you straight up place order an error is shown at the top of checkout when the coupon is removed and checkout is blocked.

However, I found when you update other data such as your address, it sends a batch request and some of the routes (update customer) do not surface errors in the UI. This is due to a bug in `receiveCartContents`.

We'll need to ensure that a batch request does not go through like this, or ensure the checkout route is never batched.
